src/Frontend/src/components/MaximizableCodeEditor.vue

```diff
@@ -0,0 +1,207 @@
+<script setup lang="ts">
+import { ref, onMounted, onBeforeUnmount } from "vue";
+import CodeEditor from "@/components/CodeEditor.vue";
+import DiffMaximizeIcon from "@/assets/diff-maximize.svg";
+import DiffCloseIcon from "@/assets/diff-close.svg";
+import { Extension } from "@codemirror/state";
+import { CodeLanguage } from "@/components/codeEditorTypes";
+
+const modelValue = defineModel<string>({ required: true });
+
+withDefaults(
+  defineProps<{
+    language?: CodeLanguage;
+    readOnly?: boolean;
+    showGutter?: boolean;
+    ariaLabel?: string;
+    extensions?: Extension[];
+    modalTitle?: string;
+  }>(),
+  {
+    readOnly: false,
+    showGutter: false,
+    extensions: () => [],
+    modalTitle: "Code View",
+  }
+);
+
+// Component state for maximize functionality
+const showMaximizeModal = ref(false);
+const showMaximizeButton = ref(false);
+
+// Handle maximize functionality
+const toggleMaximizeModal = () => {
+  showMaximizeModal.value = !showMaximizeModal.value;
+};
+
+// Handle mouse enter/leave for showing maximize button
+const onEditorMouseEnter = () => {
+  showMaximizeButton.value = true;
+};
+
+const onEditorMouseLeave = () => {
+  showMaximizeButton.value = false;
+};
+
+// Handle ESC key to close modal
+const handleKeyDown = (event: KeyboardEvent) => {
+  if (event.key === "Escape" && showMaximizeModal.value) {
+    showMaximizeModal.value = false;
+  }
+};
+
+// Setup keyboard events for maximize modal
+onMounted(() => {
+  window.addEventListener("keydown", handleKeyDown);
+});
+
+// Clean up event listeners when component is destroyed
+onBeforeUnmount(() => {
+  window.removeEventListener("keydown", handleKeyDown);
+});
+</script>
+
+<template>
+  <div class="code-editor-wrapper" @mouseenter="onEditorMouseEnter" @mouseleave="onEditorMouseLeave">
+    <!-- Regular CodeEditor -->
+    <div class="editor-container">
+      <!-- Maximize Button (shown on hover) -->
+      <button v-if="showMaximizeButton" @click="toggleMaximizeModal" class="maximize-button" title="Maximize view">
```

Comment:
```suggestion
      <button v-if="showMaximizeButton" @click="toggleMaximizeModal" class="maximize-button" v-tippy="`Maximize view`">
```

Reply:
@cquirosj : It will good to decrease the size of the modal window
- components\messages2\DiffViewer.vue
The following change in .maximize-modal-content CSS decreases the window slightly
.maximize-modal-content {
  background-color: white;
  width: calc(100% - 200px);
  height: calc(100% - 200px);
  border-radius: 4px;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
}